### PR TITLE
Add Google Contacts integration

### LINKED ISSUE #53: CardDAV integration
# Summary
When using names or organisations as recipients, do a search in the senders CardDAV contact book to find more details.

# Examples

# Context

### PR DESCRIPTION
**Pull request by widforss**
# Proposed changes
### Misc
* Moved out everything Google to its own module, with a cleaner interface to use inside automation modules.
* Use the util `get_email` in send.py as well.
* Also replaced send.py's multiple contact match handling with schedule.py's, as it is better.

### Google Contacts
* Checks Gmails "Other contacts" to see if any of the unknown contacts in `lib.utils.contacts.get_emails` have a match there.
* Before it adds such a contact to the list, it asks the user about it specifically.

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

- [x] "schedule meeting with aron@antarkt.is at 20:00" accepted the email as recipient
- [x] "schedule meeting with John at 20:00" asked about the different Johns
- [x] "schedule meeting with Erik at 20:00" asked about Erik.Elfgren@ltu.se and added him as recipient
- [x] The same tests but using the email module

# Related issue
Resolves #53

### CODE REVIEW

**Comment by widforss** [comment]
This is on hold until #73 and #74 are merged.

**Review by 97gushan — APPROVED** [review]

**Comment by 97gushan** [comment]
remember to not use real email addresses of outside people when testing, we shouldn't clutter other peoples calendars

**Comment by widforss** [comment]
Yeah, that was a mistake. It carried over from mail, which was tested local-only, but Erik accidentally got a meeting invite …

File: lib/automate/google.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        """
+        The file token.pickle stores the user's access and refresh tokens, and is
+        created automatically when the authorization flow completes for the first
+        time.
+        """
+        old_dir = os.getcwd()
+        os.chdir(os.path.dirname(os.path.abspath(__file__)) + "/../..")
+        self.username = username
+        creds = None
+        pickle_filename = f"{username}_token.pickle"
```
Comment (widforss) [inline lib/automate/google.py:32]: This should be in the pickle jar.

File: lib/automate/google.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+class People:
+    def __init__(self, google: Google, username):
+        self.username = username
+        self.google = google
+        self.service = build("people", "v1", credentials=google.creds)
+        self.contacts = None
+        self.fetch()
+
+    def fetch(self):
+        pickle_filename = f"{self.username}_contacts.pickle"
```
Comment (widforss) [inline lib/automate/google.py:116]: This should be in the pickle jar.

**Review by 97gushan — APPROVED** [review]

**Review by markhakansson — APPROVED** [review]

**Review by 97gushan — APPROVED** [review]

**Pull request merged**